### PR TITLE
Display an error banner on network error on start-/stop work..

### PR DESCRIPTION
**Pull request by pinkiebell**
...instead of showing nothing.

Related #1678

### CODE REVIEW

File: app/assets/v2/js/shared.js [hunk 1]
```diff
@@ -266,7 +266,11 @@ var mutate_interest = function(bounty_pk, direction, data) {
     }
     return false;
   }).fail(function(result) {
-    alert(result.responseJSON.error);
+    _alert('Network or API error. Please reload the page and try again.', 'error');
```
Comment (SaptakS) [inline app/assets/v2/js/shared.js]: Please use `gettext()` to ensure translation

**Comment by owocki** [comment]
> Display an error banner on network error on start-/stop work instead of showing nothing

I wasnt around to review this before.. but it seems clear to me from before this PR that there was an `alert()` that displayed the error before.  Why does the title of this ticket allege 'showing nothing' ?

**Comment by pinkiebell** [comment]
@owocki 
We failed silently on network errors before.
In case of API errors, well you are right, we should show the error message to the user.
The reason I did a `console.log` was because that happened only on 'glitches'.
At least I assumed that back then ;)

**Comment by owocki** [comment]
> We failed silently on network errors before.

Maybe we should check and see if the response has any error code in it and if it does, show an alert with the error message in it.  If not, show a generic error msg

**Comment by pinkiebell** [comment]
@owocki #1918 

@PixelantDesign 
The red banner with possible messages:
            'error': _('You must be authenticated via github to use this feature!'),
            'error': _(f'There is already someone working on this bounty.'),
            'error': _(f'You may only work on max of {num_issues} issues at once.'),
            'error': _('Because a staff member has had to remove you from a bounty in the past, you are unable to start'
                       'more work at this time. Please leave a message on slack if you feel this message is in error.'),
            'error': _('You have already started work on this bounty!'),

**Pull request merged**